go/Connection/TypeDBDatabaseImpl.go

```diff
@@ -0,0 +1,87 @@
+package Connection
+
+import (
+	"typedb_driver/go/api/database"
```

Comment:
You're still using tabs as characters! It is stinky and not pretty. 

Moreover, I haven't seen an answer to my question about code conventions: how many spaces should we use for indents in `go`?

Reply:
The convention is 1 tab, or 4 spaces I may be getting auto-formatted at times and missing it.